src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -852,13 +849,17 @@ static bool tcp_flush(grpc_tcp* tcp, grpc_error** error) {
     msg.msg_iov = iov;
     msg.msg_iovlen = iov_size;
     msg.msg_flags = 0;
+    bool tried_sending_message = false;
     if (tcp->outgoing_buffer_arg != nullptr) {
-      if (!tcp_write_with_timestamps(tcp, &msg, sending_length, &sent_length,
-                                     error)) {
+      if (!tcp_write_with_timestamps(tcp, &msg, sending_length, &sent_length)) {
```

Comment:
should we cache this and not try again if the platform does not support?

Reply:
done

Reply:
I made it per fd for now. In the future if we are sure that is not per fd rather for the entire platform, then we can have a process wide flag